console/src/main/scripts/plugins/metrics/ts/triggers/eventTrigger.ts

```diff
@@ -33,15 +33,25 @@ module HawkularMetrics {
 
           this.adm.trigger = {};
           // updateable
-          this.adm.trigger['description'] = triggerData.trigger.description;
           this.adm.trigger['enabled'] = triggerData.trigger.enabled;
-          this.adm.trigger['name'] = triggerData.trigger.name;
           this.adm.trigger['severity'] = triggerData.trigger.severity;
 
-          this.adm.trigger['email'] = triggerData.trigger.actions.email[0];
+          if ( triggerData.trigger.actions !== undefined ) {
+            triggerData.trigger.actions.forEach((triggerAction: ITriggerAction) => {
+              this.adm.trigger[triggerAction.actionPlugin] = triggerAction.actionId;
+            });
+          }
+          if ( this.adm.trigger['email'] === undefined || this.adm.trigger['email'] === null ) {
+            this.adm.trigger['emailEnabled'] = false;
+            this.adm.trigger['email'] = this.$rootScope.userDetails.email;
+          } else {
+            this.adm.trigger['emailEnabled'] = true;
+          }
 
           // presentation
           this.adm.trigger['context'] = triggerData.trigger.context;
+          this.adm.trigger['description'] = triggerData.trigger.description;
```

Comment:
In the future, try not to use **trigger.['description']** because it bypasses type checking, but instead use **trigger.description** (which the IDE will also autocomplete for you).